EcommerceStore.Application/Services/BrandService.cs

```diff
@@ -0,0 +1,108 @@
+using EcommerceStore.Application.Exceptions;
+using EcommerceStore.Application.Interfaces;
+using EcommerceStore.Application.Models.InputModels;
+using EcommerceStore.Application.Models.ViewModels;
+using EcommerceStore.Domain.Entities;
+using EcommerceStore.Domain.Interfaces;
+
+namespace EcommerceStore.Infrastucture.Services
+{
+    public class BrandService : IBrandService
+    {
+        private readonly IBrandRepository _brandRepository;
+
+        public BrandService(IBrandRepository brandRepository)
+        {
+            _brandRepository = brandRepository;
+        }
+
+        public async Task CreateBrandAsync(BrandInputModel brandIm)
+        {
+            var existingBrand = await _brandRepository.GetByNameAsync(brandIm.Name);
+
+            if (existingBrand.Name is null)
```

Comment:
1. This code will throw null ref exception if existing brand is not null, as you are trying to access Name property of null, you should check if existing brand is null
2. please throw the exception at the top if brand is not null, in order to get rid of long if block(this comment relates to all similar places)